src/storage/btree.c

```diff
@@ -14339,8 +14364,9 @@ btree_find_rightmost_leaf (THREAD_ENTRY * thread_p, BTID * btid, VPID * pg_vpid,
  * Note: Find the page identifier for the first/last leaf page of the B+tree index.
  */
 static PAGE_PTR
-btree_find_boundary_leaf (THREAD_ENTRY * thread_p, BTID * btid, VPID * pg_vpid, BTREE_STATS * stat_info,
-			  BTREE_BOUNDARY where)
+btree_find_boundary_leaf_with_repl_desync_check (THREAD_ENTRY * thread_p, const BTID * btid, VPID * pg_vpid,
+						 BTREE_STATS * stat_info, const BTREE_BOUNDARY where,
+						 bool & desync_occured)
```

Comment:
function should also reset `desync_occured` - just to make sure - even if reset outside currently